koch.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/sh
+
+# Script to bootstrap `koch` if necessary
+#
+# Copyright (c) 2021 Leorize <leorize+oss@disroot.org>
+#
+# Licensed under the terms of the MIT license. See "license.txt" included in
+# this distribution for more information.
+
+# Exit on usage of undefined variable
+set -u
+# Exit if any command errors
+set -e
+
+# Print and run the input command
+run() {
+  echo "Running:" "$@"
+  "$@"
+}
+
+# The directory where this script resides
+baseDirectory=$(dirname "$0")
+
+# The bootstrap compiler path.
+bootstrapCompiler="$baseDirectory/bin/nim_boot"
+
+# The bootstrap compiler source.
+bootstrapSource="$baseDirectory/build/csources"
+
+# The nimcache folder.
+nimcache="$baseDirectory/nimcache"
+
+# Build config
+buildConfig="$baseDirectory/config/build_config.txt"
+
+# Get number of processing cores. Returns 1 if this number can not be retrieved.
+get_num_cores() {
```

Comment:
I thought the typical recommendation was for `min(RAM/2GB, 2 x CPU-threads)`? With the former winning over the latter if lazy.

Reply:
CPU threads is several order of magnitude easier to get (in a cross-platform manner) :P

It keeps things simple and that's what most builder uses nowadays.